### PR TITLE
add ndt_scan_matcher readme

### PR DESCRIPTION
**Pull request by YamatoAndo**
## Description
add ndt_scan_matcher readme

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [ ] Read [pull request guidelines][pull-request-guidelines]
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets
- [ ] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s). Found false positives? See the [guidelines][clang-tidy-guidelines].
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[clang-tidy-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/ClangTidyGuideline/
[coding-guidelines]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/1194394777/T4
[pull-request-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/PullRequestGuideline/
[release-notes]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/563774416
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

File: localization/pose_estimator/ndt_scan_matcher/ndt_scan_matcher/README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# ndt scan matcher
```
Comment (kmiya) [inline localization/pose_estimator/ndt_scan_matcher/ndt_scan_matcher/README.md:1]:
```suggestion
# ndt_scan_matcher
```
Nit: Should be snake_case. Please check the merge guidelines <https://tier4.atlassian.net/wiki/spaces/ATWROSS/pages/2163574164>.

**Review by yukkysaito — APPROVED** [review]

File: localization/pose_estimator/ndt_scan_matcher/ndt_scan_matcher/README.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+## Purpose
+
+ndt_scan_matcher is a package for position estimation using the NDT scan matching method.
+
+There are two main functions of this package :
```
Comment (kmiya) [inline localization/pose_estimator/ndt_scan_matcher/ndt_scan_matcher/README.md:7]:
```suggestion
There are two main functions of this package:
```
Nit: the space can be removed.

File: localization/pose_estimator/ndt_scan_matcher/ndt_scan_matcher/README.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+### Output
+
+| Name                              | Type                                            | Description                                                                                                                              |
+| --------------------------------- | ----------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------- |
+| `ndt_pose`                        | `geometry_msgs::msg::PoseStamped`               | estimated pose                                                                                                                           |
+| `ndt_pose_with_covariance`        | `geometry_msgs::msg::PoseWithCovarianceStamped` | estimated pose with covariance                                                                                                           |
+| `/diagnostics`                    | `diagnostic_msgs::msg::DiagnosticArray`         | diagnostics                                                                                                                              |
```
Comment (kmiya) [inline localization/pose_estimator/ndt_scan_matcher/ndt_scan_matcher/README.md:28]:
```suggestion
| `diagnostics`                    | `diagnostic_msgs::msg::DiagnosticArray`         | diagnostics                                                                                                                              |
```
Question: could the slash be removed? I'm no sure.

File: localization/pose_estimator/ndt_scan_matcher/ndt_scan_matcher/README.md (hunk 1 above)
Comment (YamatoAndo) [reply localization/pose_estimator/ndt_scan_matcher/ndt_scan_matcher/README.md:1]: Thank you!

File: localization/pose_estimator/ndt_scan_matcher/ndt_scan_matcher/README.md (hunk 2 above)
Comment (YamatoAndo) [reply localization/pose_estimator/ndt_scan_matcher/ndt_scan_matcher/README.md:7]: Thank you!

File: localization/pose_estimator/ndt_scan_matcher/ndt_scan_matcher/README.md (hunk 3 above)
Comment (kenji-miyake) [reply localization/pose_estimator/ndt_scan_matcher/ndt_scan_matcher/README.md:28]: `/diagnostics` and `/diagnostics_agg` are the special cases.
They should be the global namespace.

Comment (YamatoAndo) [reply localization/pose_estimator/ndt_scan_matcher/ndt_scan_matcher/README.md:28]: This topic is intentionally named with `/` as a global topic name.

https://github.com/tier4/AutowareArchitectureProposal.iv/blob/use-autoware-auto-msgs/localization/pose_estimator/ndt_scan_matcher/ndt_scan_matcher/src/ndt_scan_matcher_core.cpp#L150

Comment (kenji-miyake) [reply localization/pose_estimator/ndt_scan_matcher/ndt_scan_matcher/README.md:28]: FYI https://github.com/ros/diagnostics/blob/9db8532c83969d2723c83f188a130100c2b55551/diagnostic_aggregator/src/aggregator.cpp#L69-L70

Comment (kmiya) [reply localization/pose_estimator/ndt_scan_matcher/ndt_scan_matcher/README.md:28]: Thank you for the information! Resolve this.

File: localization/pose_estimator/ndt_scan_matcher/ndt_scan_matcher/README.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
+# ndt_scan_matcher
+
+## Purpose
+
+ndt_scan_matcher is a package for position estimation using the NDT scan matching method.
+
+There are two main functions of this package:
```
Comment (IshitaTakeshi) [inline localization/pose_estimator/ndt_scan_matcher/ndt_scan_matcher/README.md:7]: Maybe `There are two main functions in this package`?

File: localization/pose_estimator/ndt_scan_matcher/ndt_scan_matcher/README.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+- estimate position by scan matching
+- estimate initial position via ros service using Monte Carlo method
```
Comment (IshitaTakeshi) [inline localization/pose_estimator/ndt_scan_matcher/ndt_scan_matcher/README.md:10]: Maybe `the Monte Carlo method`?

File: localization/pose_estimator/ndt_scan_matcher/ndt_scan_matcher/README.md (hunk 4 above)
Comment (YamatoAndo) [reply localization/pose_estimator/ndt_scan_matcher/ndt_scan_matcher/README.md]: Thank you! https://github.com/tier4/AutowareArchitectureProposal.iv/pull/690/commits/bb61812d80b259692c01e35b2d082c315696ef5f

File: localization/pose_estimator/ndt_scan_matcher/ndt_scan_matcher/README.md (hunk 5 above)
Comment (YamatoAndo) [reply localization/pose_estimator/ndt_scan_matcher/ndt_scan_matcher/README.md]: Thank you! https://github.com/tier4/AutowareArchitectureProposal.iv/pull/690/commits/bb61812d80b259692c01e35b2d082c315696ef5f

File: localization/pose_estimator/ndt_scan_matcher/ndt_scan_matcher/README.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
+# ndt_scan_matcher
+
+## Purpose
+
+ndt_scan_matcher is a package for position estimation using the NDT scan matching method.
+
+There are two main functions in this package:
+
+- estimate position by scan matching
+- estimate initial position via ros service using the Monte Carlo method
```
Comment (IshitaTakeshi) [inline localization/pose_estimator/ndt_scan_matcher/ndt_scan_matcher/README.md:10]: I think it is better to write `the ROS service`

Comment (YamatoAndo) [reply localization/pose_estimator/ndt_scan_matcher/ndt_scan_matcher/README.md]: OK! Thank you! https://github.com/tier4/AutowareArchitectureProposal.iv/pull/690/commits/dc74879f1c2326301855e4f44649bbba1332a692

**Review by IshitaTakeshi — APPROVED** [review]

**Pull request merged**